pkg/controller/factory/factory.go

```diff
@@ -0,0 +1,67 @@
+package factory
+
+import (
+	"time"
+
+	"k8s.io/client-go/tools/cache"
+
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+// Factory is generator that generate standard Kubernetes controllers.
+// Factory is really generic and should be only used for simple controllers that does not require special stuff..
+type Factory struct {
+	sync           SyncFunc
+	resyncInterval time.Duration
+	informers      []cache.SharedInformer
+	cachesToSync   []cache.InformerSynced
+}
+
+// New return new factory instance.
+func New() *Factory {
+	return &Factory{}
+}
+
+// Sync is used to set the controller synchronization function. This function is the core of the controller and is
+// usually hold the main controller logic.
+func (f *Factory) Sync(syncFn SyncFunc) *Factory {
+	f.sync = syncFn
+	return f
+}
+
+// Informers is used to register event handlers and get the caches synchronized functions.
+// Pass informers you want to use to react to changes on resources. If informer event is observed, then the Sync() function
+// is called.
+func (f *Factory) Informers(informers ...cache.SharedInformer) *Factory {
```

Comment:
is everything in our operators of this type? Also the dynamic informers?

Reply:
yeah, I haven't found operator with dynamic, but if there are any, we can add special knob.